python/taichi/lang/impl.py

```diff
@@ -158,6 +158,25 @@ def subscript(value, *indices):
             ])
         else:
             return Expr(_ti_core.subscript(var, indices_expr_group))
+    elif isinstance(value, AnyArray):
+        field_dim = int(value.ptr.get_attribute("dim"))
+        element_dim = len(value.element_shape)
+        if field_dim != index_dim + element_dim:
+            raise IndexError(
+                f'Field with dim {field_dim - element_dim} accessed with indices of dim {index_dim}'
+            )
+        if element_dim == 0:
+            return Expr(_ti_core.subscript(value.ptr, indices_expr_group))
+        else:
```

Comment:
nit: no need to `else` the rest of the code

Reply:
done